File: Day045_2025/Q045_Leetcode.cpp
class Q045_Leetcode {
    vector<int> store;
    int size;
public:
    Q045_Leetcode() {
        // Q. Product of the Last K Numbers
        store.push_back(1);
        size = 0;
    }
    
    void add(int num) {
        if(num == 0){
            store = {1};
            size = 0;
        }
        else{
            store.push_back(num * store[size]);
            size += 1;
        }
    }
    
    int getProduct(int k) {
        if(k > size){
            return 0;
        }
        return store[size] / store[size - k];
    }
};

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers* obj = new ProductOfNumbers();
 * obj->add(num);
 * int param_2 = obj->getProduct(k);
 */

File: Day045_2025/Q045_Leetcode.py
class ProductOfNumbers:
    def __init__(self):
        self.store = [1]
        self.size = 0

    def add(self, num: int) -> None:
        if num == 0:
            self.store = [1]
            self.size = 0
        else:
            self.store.append(num * self.store[self.size])
            self.size += 1

    def getProduct(self, k: int) -> int:
        if k > self.size:
            return 0
        return self.store[self.size] // self.store[self.size - k]

# Your ProductOfNumbers object will be instantiated and called as such:
# obj = ProductOfNumbers()
# obj.add(num)
# param_2 = obj.getProduct(k)
